app/controllers/application_controller.rb

```diff
@@ -274,15 +274,24 @@ def set_locale
     I18n.locale = LocaleChooser.new(params[:locale], request).locale
   end
 
+  def sp_session_ial
+    sp_session[:ial]
+  end
+
+  def sp_session_ial_1_or_2
+    return 1 if sp_session[:ial].blank?
+    sp_session[:ial] > 1 ? 2 : 1
+  end
+
   def increment_monthly_auth_count
     return unless current_user&.id
     issuer = sp_session[:issuer]
-    return if issuer.blank? || !first_auth_of_session?(issuer)
-    MonthlySpAuthCount.increment(current_user.id, issuer, sp_session[:ial2] ? 2 : 1)
+    return if issuer.blank? || !first_auth_of_session?(issuer, sp_session_ial)
+    MonthlySpAuthCount.increment(current_user.id, issuer, sp_session_ial_1_or_2)
```

Comment:
On second thought... I have another idea for how we can clean this up. If we pass in plain `ial` level, we can refactor the method undeneath use issuer + ial to create an `IalContext` object which has `ial2_or_greater?` which does what we want here, removing the need for `sp_session_ial_1_or_2`